### PR TITLE
DHFPROD-4382: Add matching step to flow

### PR DESCRIPTION
**Pull request by brucean52**
### Description
This can be tested by running the matching step from the matching card and seeing the results in query console

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by Sanjeevani19 — APPROVED** [review]

**Review by xnikhil08 — DISMISSED** [review]

**Review by ryanjdew — DISMISSED** [review]

**Review by ngodugu-marklogic — CHANGES_REQUESTED** [review]
@brucean52 When I select add step to a new flow, then we are getting redirected to the flow screen to enter in details about the new flow, But the match step is not getting displayed in the flow once it's created. This is working as expected when we add new flow from the Run tile. Please refer [https://drive.google.com/file/d/1ez_oZz463odHsKGYyauXXg-g2pWsLcZl/view](https://drive.google.com/file/d/1ez_oZz463odHsKGYyauXXg-g2pWsLcZl/view)

**Comment by brucean52** [comment]
> @brucean52 When I select add step to a new flow, then we are getting redirected to the flow screen to enter in details about the new flow, But the match step is not getting displayed in the flow once it's created. This is working as expected when we add new flow from the Run tile. Please refer https://drive.google.com/file/d/1ez_oZz463odHsKGYyauXXg-g2pWsLcZl/view

Fixed this.

**Review by ryanjdew — APPROVED** [review]

**Review by ngodugu-marklogic — APPROVED** [review]

File: marklogic-data-hub-central/ui/src/components/entities/matching/matching-card.test.tsx [hunk 1]
```diff
@@ -116,4 +119,161 @@ describe("Matching cards view component", () => {
...
+      expect(getByText("Run Step in an existing flow")).toBeInTheDocument();
+      userEvent.click(getByTestId("matchCustomers-run-flowsList"));
+      userEvent.click(getByLabelText("customerJSONFlow-run-option"));
+    });
+    wait(() => {
+      expect(getByLabelText("step-not-in-flow-run")).toBeInTheDocument();
+      userEvent.click(getByTestId("matchCustomers-to-test-Confirm"));
+    });
+    // Check if the /tiles/run/add route has been called
+    wait(() => { expect(mockHistoryPush).toHaveBeenCalledWith("/tiles/run/add"); });
```
Comment (ngodugu-marklogic) [inline marklogic-data-hub-central/ui/src/components/entities/matching/matching-card.test.tsx:255]: @brucean52 can we also add validations for checking whether we are getting redirected to flows and the match step is added to the existing flow/the newly added flow.

**Review by ngodugu-marklogic — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/entities/matching/matching-card.test.tsx (hunk 1 above)
Comment (ngodugu-marklogic) [reply marklogic-data-hub-central/ui/src/components/entities/matching/matching-card.test.tsx:255]: As per discussion with Bruce, will be taken care as part of DHFPROD-4548.

**Review by ngodugu-marklogic — APPROVED** [review]

File: marklogic-data-hub-central/ui/src/components/entities/matching/matching-card.test.tsx (hunk 1 above)
Comment (brucean52) [reply marklogic-data-hub-central/ui/src/components/entities/matching/matching-card.test.tsx:255]: As we discussed, there's already a RTL test for this and I will add additional e2e testing for this in a later PR.

**Pull request merged**